aloha-core/src/main/scala/com/eharmony/aloha/dataset/csv/encoding/encodings.scala

```diff
@@ -26,17 +51,37 @@ object Encoding {
 
 // TODO: Fill in
 case object ThermometerEncoding extends Encoding {
-    override def finalizer[A](sep: String, nullString: String, values: Iterable[String]) =
+    override def finalizer[A](sep: String, nullString: String, values: Seq[String]) =
+        throw new UnsupportedOperationException("ThermometerEncoding not implemented!")
+
+    override def columnarFinalizer[A](nullString: String, values: Seq[String]) =
         throw new UnsupportedOperationException("ThermometerEncoding not implemented!")
 
     override def csvHeadersForColumn(c: CsvColumn): Seq[String] =
         throw new UnsupportedOperationException("ThermometerEncoding not implemented!")
 }
 
-case object HotOneEncoding extends Encoding {
+case object HotOneEncoding extends Encoding with VectorHeaders {
+
+    private[this] final case class HotOneVec[A](ind: Map[String, Int]) extends (Option[A] => Seq[String]) {
+        private[this] val n = ind.size
+        private[this] val missing = Stream.fill(n)("0")
+
+        private[this] def densifyVector(i: Int, n: Int): Seq[String] =
+            (Stream.fill(i)("0") :+ "1") ++ Stream.fill(n - i - 1)("0")
+
+        def apply(x: Option[A]): Seq[String] = {
+            val vec = for {
+                a <- x
+                i <- ind.get(a.toString)
+            } yield densifyVector(i, n)
+            vec.getOrElse(missing)
+        }
+    }
+
     private[this] case class HotOne[A](sep: String, ind: Map[String, Int]) extends (Option[A] => String) {
         private[this] val n = ind.size
-        private[this] def missing(n: Int, sep: String) = (0 to n).mkString(sep)
```

Comment:
I assume this should have been `until`?

Reply:
No.  It turns out this bug was way worse than that.  Instead of creating a vector of zeros, **0**, it was creating the vector 

[1, 2, 3, ..., n]<sup>*T*</sup>